Dockerfile

```diff
@@ -1,17 +1,22 @@
-FROM golang:1.6-alpine
+FROM golang:1.9-alpine
 
-RUN apk add --no-cache gcc git libc-dev
+RUN apk add --no-cache gcc git libc-dev curl
 
-ENV DOCKER_MACHINE_VERSION v0.7.0
+ENV DOCKER_MACHINE_VERSION v0.12.2
+RUN git clone --depth 1 -b "$DOCKER_MACHINE_VERSION" https://github.com/docker/machine.git "$GOPATH/src/github.com/docker/machine" \
+    && go build -v -o "$GOPATH/bin/docker-machine" github.com/docker/machine/cmd
 
-RUN git clone --depth 1 -b "$DOCKER_MACHINE_VERSION" https://github.com/docker/machine.git "$GOPATH/src/github.com/docker/machine"
-RUN go build -v -o "$GOPATH/bin/docker-machine" github.com/docker/machine/cmd
-
-RUN go get -d -v github.com/joyent/gosdc/...
+ENV DEP_VERSION 0.3.2
+ENV DEP_CHECKSUM 322152b8b50b26e5e3a7f6ebaeb75d9c11a747e64bbfd0d8bb1f4d89a031c2b5
+RUN curl -Lso /tmp/dep "https://github.com/golang/dep/releases/download/v${DEP_VERSION}/dep-linux-amd64" \
+    && echo "${DEP_CHECKSUM}  /tmp/dep" | sha256sum -c \
+    && mv /tmp/dep /usr/local/bin/dep \
+    && chmod 0755 /usr/local/bin/dep
 
 WORKDIR $GOPATH/src/github.com/joyent/docker-machine-driver-triton
 COPY *.go ./
-
-RUN go install -v ./...
+COPY Gopkg.* ./
+RUN dep ensure \
+    && go install -v ./...
```

Comment:
I think we should either re-order this so we don't have to re-download all the deps for every `.go` change, or simply use the `vendor/` directory here too (which is probably better/saner).

So something like this _should_ do the trick:
```dockerfile
WORKDIR $GOPATH/src/github.com/joyent/docker-machine-driver-triton
COPY *.go vendor ./
RUN go install -v ./...
```
(and then we don't need `dep` in the `Dockerfile` at all, unless you want to use this `Dockerfile` for creating a dev environment for folks like me who don't have the `dep` tool installed on their host :smile:)

Reply:
Definitely debatable. The thought was to use this as a development environment (for folks that need it) and not simply a container to run `docker-machine`. That interpretation came from the original `Dockerfile` which built everything within the container as well.

But I can remove `dep` and just utilize the `vendor/` directory. This isn't a huge moving target. 👍